src/network.c

```diff
@@ -116,7 +113,7 @@ int full_message_availiable(int socket) {
 
   if (message_len_recv >= 4) {
     message_len = message[0];
-    return message_len_recv >= message_len + 4;
+    return message_len + 4;
```

Comment:
I think this needs an additional conditional check. You might be returning the size of the message even when a full message is not present. 
```suggestion
    if(message_len_recv >= message_len + 4) {
        return message_len + 4;
    }
```